src/commands.rs

```diff
@@ -1055,6 +1055,21 @@ struct Args {
     command: Commands,
 }
 
+/// Used for resolving aliases
+#[derive(clap::Parser, Clone, Debug)]
+struct AliasArgs {
+    #[clap(flatten)]
+    global_args: GlobalArgs,
+    #[clap(subcommand)]
+    command: AliasCommand,
+}
+
+#[derive(Subcommand, Debug, Clone)]
+enum AliasCommand {
+    #[clap(external_subcommand)]
+    Alias(Vec<String>),
+}
```

Comment:
> 1. By using a `String`, I take it you don't accept non-UTF8 or invalid UTF8 arguments?

Yes, that's correct. I know you already know a bit about jj, so in case you're also wondering *why* I made that decision, it's because VCSs are mostly used for collaborating on a project, and then you need an agreed-upon encoding for almost everything: paths, branch names, and commit message. I does mean users can't use jj for versioning their own files on local disk encoded in Shift JIS, for example.

> 2. Is there a reason you dispatch everything through aliases rather than flattening `Commands` into this?

No good, deliberate reason, no :) However, when I tried changing the `Commands` struct to have an `Alias` command, I ran into two problems:

1. I have to provide my own error message for when the specified command is not a valid alias. That's not a big deal, and I'm sure `clap` already provides some way of producing an error message that matches its style.
2. The user can no longer define aliases that override builtins. Maybe that's actually for the better, actually. I'll check with the rest of the team.

Reply:
I forgot to say earlier, but thanks for reviewing!

> 1. I have to provide my own error message for when the specified command is not a valid alias. That's not a big deal, and I'm sure `clap` already provides some way of producing an error message that matches its style.

I've figured out how to replicate all but the coloring. That's close enough for me.

> 2. The user can no longer define aliases that override builtins. Maybe that's actually for the better, actually. I'll check with the rest of the team.

We decided to disallow overriding builtins. I noticed that git also disallows it, but hg allows it.